itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorOnTezTest.java

```diff
@@ -261,22 +326,77 @@ protected void insertMmTestData(String tblName, int iterations) throws Exception
     }
 
     List<String> getAllData(String tblName) throws Exception {
-      return getAllData(null, tblName);
+      return getAllData(null, tblName, false);
     }
 
-    List<String> getAllData(String dbName, String tblName) throws Exception {
+    List<String> getAllData(String tblName, boolean withRowId) throws Exception {
+      return getAllData(null, tblName, withRowId);
+    }
+
+    List<String> getAllData(String dbName, String tblName, boolean withRowId) throws Exception {
       if (dbName != null) {
         tblName = dbName + "." + tblName;
       }
-      List<String> result = executeStatementOnDriverAndReturnResults("select * from " + tblName, driver);
+      StringBuffer query = new StringBuffer();
+      query.append("select ");
+      if (withRowId) {
+        query.append("ROW__ID, ");
+      }
+      query.append("* from ");
+      query.append(tblName);
+      List<String> result = executeStatementOnDriverAndReturnResults(query.toString(), driver);
       Collections.sort(result);
       return result;
     }
 
+    List<String> getDataWithInputFileNames(String dbName, String tblName) throws Exception {
+      if (dbName != null) {
+        tblName = dbName + "." + tblName;
+      }
+      StringBuffer query = new StringBuffer();
+      query.append("select ");
+      query.append("INPUT__FILE__NAME, a from ");
```

Comment:
I have seen issues with queries using virtual columns (INPUT__FILE__NAME, ROW__ID). The row number in the results were different with and without the virtual columns. - This is just a note, maybe no action is needed here

Reply:
Thanks a lot for the info. I will check, but I haven't seen any issues with this. Maybe because I don't check the row number here, just the file names?